<a href="https://colab.research.google.com/github/SaulVid/mat421sv/blob/main/Integration_methods.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Riemanns Integral**

Evaluating the following integral:

$ \int_{0}^{2} 10x \,dx$


In [80]:
import sympy as smp
x = smp.symbols('x',real = True)
f = 10*x
smp.integrate(f,(x,0,5))


125

Sympy is able to calculate the exact solution for
$ \int_{0}^{2} 10x \,dx$, which is 125. 

Now the exact solution can be compared to approximations using increasing step sizes.



In [81]:
def integral(f,a,b,n):
    delta_x = (b-a)/n
    summation = 0
    for i in range (1,n+1):
      summation += f(a + (i-1)*delta_x)*delta_x
    return summation

integral(lambda x: 10*x,0,5,10)

112.5

In [82]:
integral(lambda x:10*x,0,5,100)

123.75

In [83]:
integral(lambda x: 10*x,0,5,1000)

124.875

In [84]:
integral(lambda x: 10*x,0,5,10000)

124.98750000000001

In [85]:
integral(lambda x: 10*x,0,5,10000000)

124.99998750000006

Trapezoid Rule

Using the trapezoid rule to approximate integrals

Example, approximating the following integral with n=6

$ \int_{0}^{\pi} cos^2(x) \,dx$


In [86]:
import numpy as np
def f(x):
  return (np.cos(x))**2

a = 0
b = np.pi
n = 6
del_x=(b-a)/n

sum=0
for i in range(1,n):
  sum = sum + del_x*f(a+i*del_x)
trapresult = del_x/2 * (f(a)+f(b)) + sum

print(trapresult)

1.5707963267948961


In [87]:
#finding actual value using Sympy

x = smp.symbols('x',real = True)
f = smp.cos(x)**2
ans = smp.integrate(f,(x,0,np.pi))
print(ans)




1.57079632679490


In [88]:
#percent error
percent_error = abs(ans-trapresult)/ans*100
print(percent_error)

2.82715971685646e-14
